python/main.py

```diff
@@ -0,0 +1,583 @@
+        s.setArrayExpression(self.visit(node.value))
+        s.setSubscriptExpression(self.visit(node.slice))
+        debug_print(s)
+        return s
+
+    def visit_Slice(self, node):
+        debug_print(ast.dump(node))
+        raise NotImplementedError
+
+    ### COMPREHENSIONS ###
+    def visit_ListComp(self, node):
+        debug_print(ast.dump(node))
+        raise NotImplementedError
+    def visit_SetComp(self, node):
+        debug_print(ast.dump(node))
+        raise NotImplementedError
+    def visit_GeneratorExp(self, node):
+        debug_print(ast.dump(node))
+        raise NotImplementedError
+    def visit_DictComp(self, node):
+        debug_print(ast.dump(node))
+        raise NotImplementedError
+    def visit_comprehension(self, node):
+        debug_print(ast.dump(node))
+        raise NotImplementedError
+
+    ### STATEMENTS ###
+    def visit_Assign(self, node):
+        debug_print(ast.dump(node))
+        # binary operator
+        binop = BinaryOperator()
+        self.add_loc_info(node, binop)
+        binop.setOperatorCode("=")
+        lhs = ExpressionList()
+        for n in node.targets:
+            lhs.addExpression(self.visit(n))
+        binop.setLhs(lhs)
+        binop.setRhs(self.visit(node.value))
+        return binop
+
+    def visit_AnnAssign(self, node):
+        debug_print(ast.dump(node))
+        raise NotImplementedError
+    def visit_AugAssign(self, node):
+        debug_print(ast.dump(node))
+        raise NotImplementedError
+    def visit_Raise(self, node):
+        debug_print(ast.dump(node))
+        pass # TODO no cpg support?
+
+    def visit_Assert(self, node):
+        debug_print(ast.dump(node))
+        raise NotImplementedError
+    def visit_Delete(self, node):
+        debug_print(ast.dump(node))
+        raise NotImplementedError
+    def visit_Pass(self, node):
+        debug_print(ast.dump(node))
+        raise NotImplementedError
+
+    ### IMPORTS ###
+    def visit_Import(self, node):
+        debug_print(ast.dump(node))
+        imp = IncludeDeclaration()
+        self.add_loc_info(node, imp)
+        if len(node.names) != 1:
+            raise NotImplementedError
+        imp.setFilename(self.fname + node.names[0].name)
+        return imp
+
+    def visit_ImportFrom(self, node):
+        debug_print(ast.dump(node))
+        # TODO cpg fixen
+        imp = IncludeDeclaration()
+        self.add_loc_info(node, imp)
+        if node.level != 0:
+            raise NotImplementedError
+        imp.setFilename(self.fname + node.module)
+        return imp
+
+    def visit_alias(self, node):
+        debug_print(ast.dump(node))
+        raise NotImplementedError
+
+
+    ### CONTROL FLOW ###
+    def visit_If(self, node):
+        debug_print(ast.dump(node))
+        stmt = IfStatement()
+        self.add_loc_info(node, stmt)
+        stmt.setCondition(self.visit(node.test))
+        if isinstance(node.body, list):
+            body = CompoundStatement()
```

Comment:
this should be handled generically by a method that handles the body and returns a compounds statement. I guess this is needed in a couple of places